### ISSUE TITLE
Unhandled exception swallowed by Q in Javascript code

### ISSUE THREAD

**Issue by Jaifroid** [kiwix/kiwix-js#348]
I was investigating some odd behaviour I couldn't account for in Firefox, and was stepping through the code, when suddenly I ended up in Q catching an exception which I managed to trace back to our unused JavaScript parsing code. I have tested this on master, and it is there too.

The error is:
```
TypeError: Unable to get property 'match' of undefined or null reference
at Anonymous function (file:///C:/....../kiwix-js/www/js/app.js:948:17)
```
And line 948 is the last of the lines below:
```
// Load Javascript content
            $('#articleContent').contents().find('script').each(function() {
                var script = $(this);
                // We try to find its name (from an absolute or relative URL)
                var srcMatch = script.attr("src").match(regexpMetadataUrl);
```
To reproduce, put a break on that line and load up any wiki page that might have a javascript in its head. I was opening the landing page of the latest French Wiktionary. Then step over the line.

The exception gets swallowed, but leaves something in a strange state. Whether this is the actual cause of the iframe weirdness I'm investigating, I'm not sure, but I landed straight in this exception, very oddly, from the onreadystatechange code when I attempted to step over it. In any case, it's not good to have
unhandled exceptions hiding in our code. I suggest we comment out the entire Load JavaScript content block for now. Do you agree @mossroy ?

**Comment by Jaifroid** [comment]
Added a commit in new branch to deal with this: https://github.com/kiwix/kiwix-js/commit/4821952f3413d2b6fe757c5a98a9d8741bfb3f17.
Obviously it would be better to fix, but this code is currently unused, I believe. Hence I have added a note for future development.

**Comment by mossroy** [comment]
It looks like a bug in the case when a script tag has no src attribute (which is possible when you use inline javascript).
In this case, we don't have anything to inject, so I think we might simply add a test before that line, like :  
```
if (script.attr("src)) {
    // We try to find its name (from an absolute or relative URL)
    var srcMatch = script.attr("src").match(regexpMetadataUrl);
    ...
}
```

**Comment by Jaifroid** [comment]
We can fix the immediate issue here, as you suggest. However, currently the JavaScript parsing code is unused, so it makes little sense to be running readBinaryFile() on any that are found only to end in a comment that disables the insertion of the file. It adds overhead to the app, and may well contribute to slowness. It would make sense to disable this routine entirely until we work on getting JavaScript to work properly in the app. Given the popularity of PhET, we should do that sooner rather than later. In the meantime, we can either comment out the entire routine, or we can turn it into an uncalled function:
```
// Load Javascript content
    function loadJavascript() {
        $('#articleContent').contents().find('script').each(function () {
            var script = $(this);
             .....
```
I'd actually be in favour of making functions out of the three routines we have for CSS, images and JavaScript. It will allow us, later on, to do some simple control of the order of loading, if we need to (I know it's not a simple matter of running them sequentially).

**Comment by mossroy** [comment]
I agree.

In this case, we should name these functions so that it's clear that they are specific to the jQuery mode. Like "loadJavascriptWithJQuery"

**Comment by Jaifroid** [comment]
Fixed with #349.
